integration-tests/narayana-jta/src/main/java/io/quarkus/narayana/jta/TransactionBeanWithEvents.java

```diff
@@ -0,0 +1,139 @@
+import javax.inject.Inject;
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+import javax.transaction.SystemException;
+import javax.transaction.Transaction;
+import javax.transaction.TransactionManager;
+import javax.transaction.TransactionScoped;
+import javax.transaction.Transactional;
+
+import org.jboss.logging.Logger;
+
+@ApplicationScoped
+public class TransactionBeanWithEvents {
+    private static final Logger log = Logger.getLogger(TransactionBeanWithEvents.class);
+
+    private static int initializedCount, beforeDestroyedCount, destroyedCount;
+    private static int commitCount, rollbackCount;
+
+    @Inject
+    private TransactionManager tm;
+
+    static int getInitialized() {
+        return initializedCount;
+    }
+
+    static int getBeforeDestroyed() {
+        return beforeDestroyedCount;
+    }
+
+    static int getDestroyed() {
+        return destroyedCount;
+    }
+
+    static int getCommited() {
+        return commitCount;
+    }
+
+    static int getRolledBack() {
+        return rollbackCount;
+    }
+
+    @Transactional
+    void doInTransaction(boolean isCommit) {
+        log.debug("Running transactional bean method");
+
+        try {
+            tm.getTransaction().registerSynchronization(new Synchronization() {
+                @Override
+                public void beforeCompletion() {
+                }
+
+                @Override
+                public void afterCompletion(int status) {
+                    if (status == Status.STATUS_ROLLEDBACK) {
+                        rollbackCount++;
+                    } else if (status == Status.STATUS_COMMITTED) {
+                        commitCount++;
+                    } else {
+                        throw new IllegalStateException("Expected commit or rollback on transaction synchronization callback");
+                    }
+                }
+            });
+        } catch (Exception e) {
+            throw new IllegalStateException("Cannot get transaction to register synchronization on bean call", e);
+        }
+
+        if (!isCommit) {
+            throw new RuntimeException("Rollback here!");
+        }
+    }
+
+    void transactionScopeActivated(@Observes @Initialized(TransactionScoped.class) final Object event,
+            final BeanManager beanManager) throws SystemException {
+        Transaction tx = tm.getTransaction();
+        if (tx == null) {
+            log.error("@Intialized expects an active transaction");
+            throw new IllegalStateException("@Intialized expects an active transaction");
+        }
+        if (tx.getStatus() != Status.STATUS_ACTIVE) {
+            log.error("@Initialized expects transaction is Status.STATUS_ACTIVE");
+            throw new IllegalStateException("@Initialized expects transaction is Status.STATUS_ACTIVE");
+        }
+        Context ctx = beanManager.getContext(TransactionScoped.class);
+        if (!ctx.isActive()) {
```

Comment:
`beanManager.getContext()` throws `ContextNotActiveException` if the given context is not active...